test/functional/p2p_feefilter.py

```diff
@@ -48,6 +37,11 @@ def on_inv(self, message):
             if (i.type == MSG_TX) or (i.type == MSG_WTX):
                 self.txinvs.append(hashToHex(i.hash))
 
+    def wait_for_invs_to_match(self, test_framework, invs_expected):
+        invs_expected.sort()
+        test_framework.wait_until(lambda: invs_expected == sorted(self.txinvs),
```

Comment:
why not self.wait_until?

Reply:
Good idea, this makes the code a lot shorter -- `test_framework` doesn't have to be passed to the inv-wait function and also setting the lock parameter is not needed anymore.
Done.